File: manraj/maximum-depth-of-binary-tree.py
class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        if root:
            ldepth = self.maxDepth(root.left)
            rdepth = self.maxDepth(root.right)

            return max(ldepth + 1, rdepth + 1)
        
        return 0

File: cole/day12_maximum-depth-of-binary-tree/maximum-depth-of-binary-tree.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
int maxDepth(struct TreeNode *root)
{
    int a, b;
    return root ? ((a = maxDepth(root->left)) > (b = maxDepth(root->right)) ? a : b) + 1 : 0;
}